Gambot.Modules.Roll/RollResponseProducer.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Gambot.Core;
+using Gambot.Data;
+
+namespace Gambot.Modules.Roll
+{
+    public class RollResponseProducer : IMessageProducer
+    {
+        private IDataStore factoidDataStore;
+        private readonly IVariableHandler variableHandler;
+
+        private const string DefaultRollSuccessReply =
+            "<reply> $who rolled a $diceRoll.";
```

Comment:
I appreciate the option to customize the roll response message, but it could be a problem. It can be useful to know the individual die rolls. For example, in many games rolling a 20 is a "critical success." In Mutants & Masterminds, players with the Luck skill can re-roll if they choose, so seeing the roll before modifiers is good. However, there could be an arbitrary number of dice in an expression. "2d20" has 2 dice, "1d10+2d6" has three dice, etc. Right now our factoid syntax isn't powerful enough to represent a variable number of...erm, variables, so we may need to sacrifice the ability to customize the response in favor of building a more verbose response in code.